test/e2e/config_test.go

```diff
@@ -794,6 +800,10 @@ func TestUserWorkloadMonitorPrometheusK8Config(t *testing.T) {
 			name: "assert remote write url value in set in CR",
 			f:    assertRemoteWriteWasSet(f.UserWorkloadMonitoringNs, crName, "https://test.remotewrite.com/api/write"),
 		},
+		{
+			name: "assert query log file value is set and correct",
```

Comment:
Should we have a test that validates that after query_log_file is set, and then reset, the queryLogFile becomes `""` ?

Reply:
personally I don' think so, because we don't do it for any other config option and not entirely sure we should treat this any differently.

e2e tests are expensive time wise, and I feel we already rely on them more heavily than we should. Testing the happy path in e2e tests is sufficient in my mind unless its critical we cover other cases.

Having said that, lets leave it open and see what others think.

Reply:
Do we require a e2e test for this ? Can't this be done in the `manifest_test.go` itself?

Reply:
It's already a path that is tested. The pattern as of now in regards to config is that we test the happy path as is the case with this test.

I did already add a unit test in `manifest_test.go` for same.